demo/repo/Jenkinsfile

```diff
@@ -17,26 +17,31 @@ parallel(longerTests: {
     runTests(servers, 20)
 })
 
-stage name: 'Staging', concurrency: 1
-node {
-    servers.deploy 'staging'
-}
+milestone ordinal: 1
+lock (resource: 'staging-server', inversePrecedence: true) {
+    stage name: 'Staging'
```

Comment:
> the build is not really in _Staging_ until it acquires the lock

Well the build is certainly not in _QA_. To my mind the original behavior, of starting to show the _Staging_ table cell but pausing there, is more informative, though `PauseAction` would make it clearer.

What we get when we add `label` is another question—we have not yet decided how to render sections of the build that are not inside a labeled block, if there are any. I still feel that we would want `label` outside `lock`, with a `PauseAction` coming from lock, to make it clear that we are _trying_ to stage and just being held up.